src/dc_tools.rs

```diff
@@ -934,4 +944,10 @@ mod tests {
         assert_eq!(w, 100);
         assert_eq!(h, 50);
     }
+
+    #[test]
+    fn test_improve_single_line_input() {
+        assert_eq!(improve_single_line_input("Hi\naiae "), "Hi aiae");
+        assert_eq!(improve_single_line_input("\r\nahte\n\r"), "ahte");
```

Comment:
i'd add some test that contain linebreaks inside the string, check for tabs as well as for some unicode-non-breaking stuff.